daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dpath/NodeInfo.scala

```diff
@@ -686,8 +692,9 @@ object NodeInfo extends Enum {
           case _ => true
         }
       }
-
-      override val width: MaybeInt = MaybeInt.Nope
+      override val isSigned: Boolean = true
+      override val minWidth: MaybeInt = MaybeInt(2)
```

Comment:
Decimal is a bit of a weird case. It can be either signed or unsigned depending on the `dfdl:decimalSigned` property. And if that is false, then minWidth could be 1. Maybe we change this to a Nope to make it clear asking the minWidth of decimal doesn't really make sense, and you need to special case width checking fro decimals?

And actually, the spec says that the minimum width for xs:decimal is 8 bits, because lengthUnits="bits" is not allowed for xs:decimal. But we've removed that restriction for all the types so maybe this doesn't apply for us.